### ISSUE TITLE
Add logic mask for choosing distinct output

### ISSUE THREAD

**Issue by piotrmwojcik** [piotrmwojcik/Ptr-Net#2]
Currently, the code may return non-distinct sequence, e.g. `[1, 2, 4, 5, 6, 7, 8, 9, 9]` 

The following line should be modified so that a mask disables multiple choices of one variable.

 `ret.append(torch.argmax(logits).item())`
